lib/elixir/lib/map.ex

```diff
@@ -117,6 +117,27 @@ defmodule Map do
   @type value :: any
   @compile {:inline, fetch: 2, fetch!: 2, get: 2, put: 3, delete: 2, has_key?: 2, replace!: 3}
 
+  # TODO: Remove conditional on Erlang/OTP 24+
+  @compile {:no_warn_undefined, {:maps, :from_keys, 2}}
+  @doc """
+  Builds a map from the given `keys` and the fixed `value`.
+
+  ## Examples
+
+      iex> Map.from_keys([1, 2, 3], :number)
+      %{1 => :number, 2 => :number, 3 => :number}
+
+  """
+  @doc since: "1.14.0"
+  @spec from_keys([key], value) :: map
+  def from_keys(keys, value) do
+    if function_exported?(:maps, :from_keys, 2) do
```

Comment:
Usually we just wait until we require the OTP version exclusively. What's the reason for the `function_exported?` check in this case?

Reply:
By the time the new Elixir releasae is out, we will support 23, 24, 25, and people will most likely be running not on 23, and the perf benefits are considerable, so I thought I would squeeze it in and clean up the code.

Also, I would say it is ok to add functions if the we can trivially implement a fallback. it has also happened for Application.put_all_env in the past. :)